Installation issue: Umpire

`Umpire` build fails using `spack@develop` (7-Aug-2020, 3978db91dc4385b492b8dcbc1f2b87326f217b1b) on:
* Ubuntu 18.04 x86_64, `%gcc@7.5.0`
* Ubuntu 18.04 ppc64le, `%gcc@7.5.0`
* RHEL 7 x86_64, `%gcc@8.1.0`
* RHEL 7 ppc64le, , `%gcc@8.1.0`

Concrete spec: [umpire.spec.yaml.txt](https://github.com/spack/spack/files/5044716/umpire.spec.yaml.txt)
Build log: [umpire-build-out.txt](https://github.com/spack/spack/files/5044715/umpire-build-out.txt)
CMake Output: [Umpire_CMakeOutput.log.txt](https://github.com/spack/spack/files/5044722/Umpire_CMakeOutput.log.txt)


```
$> spack install -f ./umpire.spec.yaml
...
==> umpire: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/umpire-3.0.0-sdhtrawfgvqp7nemw5v62ch3k7pgximf' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/umpire-3.0.0-sdhtrawfgvqp7nemw5v62ch3k7pgximf/lib;/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/umpire-3.0.0-sdhtrawfgvqp7nemw5v62ch3k7pgximf/lib64' '-DCMAKE_PREFIX_PATH:STRING=/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cmake-3.17.3-5srvsj7hfqq2os2ybrj2bsgbufpgrj7m' '-DENABLE_CUDA=Off' '-DENABLE_C=On' '-DENABLE_FORTRAN=Off' '-DENABLE_NUMA=Off' '-DENABLE_OPENMP=Off' '-DBUILD_SHARED_LIBS=On' '-DENABLE_TESTS=Off' '/tmp/root/spack-stage/spack-stage-umpire-3.0.0-sdhtrawfgvqp7nemw5v62ch3k7pgximf/spack-src'

1 error found in build log:
     45    -- Standard C++11 selected
     46    -- Enabling all compiler warnings on all targets.
     47    -- Fortran support disabled.
     48    -- CMAKE_C_FLAGS flags are:    -Wall -Wextra
     49    -- CMAKE_CXX_FLAGS flags are:       -Wall -Wextra
     50    -- CMAKE_EXE_LINKER_FLAGS flags are:
  >> 51    CMake Error at blt/thirdparty_builtin/CMakeLists.txt:129 (message):
     52      ENABLE_BENCHMARKS requires ENABLE_TESTS to be ON
     53
     54
     55    -- Configuring incomplete, errors occurred!
     56    See also "/tmp/root/spack-stage/spack-stage-umpire-3.0.0-sdhtrawfgvqp7nemw5v62ch3k7pgximf/spack-build/CMakeFiles/CMakeOutput.log".
```

@cyrush @adrienbernede @rblake @adamjstewart @scottwittenburg

--- a/var/spack/repos/builtin/packages/umpire/package.py
+++ b/var/spack/repos/builtin/packages/umpire/package.py
@@ -45,6 +45,8 @@ class Umpire(CMakePackage, CudaPackage):
     variant('openmp', default=False, description='Build with OpenMP support')
     variant('deviceconst', default=False,
             description='Enables support for constant device memory')
+    variant('tests', default='none', values=('none', 'basic', 'benchmarks'),
+            multi=False, description='Tests to run')
 
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', when='+cuda', type='build')
@@ -88,7 +90,10 @@ class Umpire(CMakePackage, CudaPackage):
         options.append('-DBUILD_SHARED_LIBS={0}'.format(
             'On' if '+shared' in spec else 'Off'))
 
+        options.append('-DENABLE_BENCHMARKS={0}'.format(
+            'On' if 'tests=benchmarks' in spec else 'Off'))
+
         options.append('-DENABLE_TESTS={0}'.format(
-            'On' if self.run_tests else 'Off'))
+            'Off' if 'tests=none' in spec else 'On'))
 
         return options

